File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.kt
class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        val last = mutableMapOf<Char, Int>()
        var res = 0; var start = 0
        for ((i, c) in s.withIndex()) {
            start = maxOf(start, (last[c]?.plus(1)) ?: start)
            res = maxOf(res, i - start + 1)
            last[c] = i
        }
        return res
    }
}

File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.go
func lengthOfLongestSubstring(s string) int {
    last := make(map[rune]int)
    res, start := 0, 0
    for i, c := range s {
        if idx, ok := last[c]; ok && idx >= start {
            start = idx + 1
        }
        if i - start + 1 > res {
            res = i - start + 1
        }
        last[c] = i
    }
    return res
}